src/System.Text.Json/tests/JsonObjectTests.cs

```diff
@@ -0,0 +1,621 @@
+        public static void TestAddingKeyValuePair()
+        {
+            var employees = new JsonObject
+            {
+                EmployeesDatabase.GetNextEmployee(),
+                EmployeesDatabase.GetNextEmployee(),
+                EmployeesDatabase.GetNextEmployee(),
+                EmployeesDatabase.GetNextEmployee(),
+            };
+
+            CheckEmployeesAreDifferent(employees);
+        }
+
+        [Fact]
+        public static void TestAddingKeyValuePairAfterInitialization()
+        {
+            var employees = new JsonObject();
+            foreach (KeyValuePair<string, JsonNode> employee in EmployeesDatabase.GetTenBestEmployees())
+            {
+                employees.Add(employee);
+            }
+
+            CheckEmployeesAreDifferent(employees);
+        }
+
+        [Fact]
+        public static void TestAddingKeyValuePairsCollection()
+        {
+            var employees = new JsonObject(EmployeesDatabase.GetTenBestEmployees());
+
+            CheckEmployeesAreDifferent(employees);
+        }
+
+        [Fact]
+        public static void TestAddingKeyValuePairsCollectionAfterInitialization()
+        {
+            var employees = new JsonObject();
+            employees.AddRange(EmployeesDatabase.GetTenBestEmployees());
+
+            CheckEmployeesAreDifferent(employees);
+        }
+
+        [Fact]
+        public static void TestContains()
+        {
+            var person = new JsonObject
+            {
+                { "name", "John" },
+                { "ssn", "123456789" },
+            };
+
+            Assert.True(person.ContainsProperty("ssn"));
+            Assert.Equal("123456789", (JsonString)person["ssn"]);
+            Assert.False(person.ContainsProperty("surname"));
+        }
+
+        [Fact]
+        public static void TestAquiringAllValues()
+        {
+            var employees = new JsonObject(EmployeesDatabase.GetTenBestEmployees());
+            ICollection<JsonNode> employeesWithoutId = employees.PropertyValues;
+
+            Assert.Equal(10, employees.PropertyNames.Count);
+            Assert.Equal(10, employees.PropertyValues.Count);
+
+            foreach (JsonNode employee in employeesWithoutId)
+            {
+                Assert.IsType<JsonObject>(employee);
+            }
+        }
+
+        [Fact]
+        public static void TestReplacingsonObjectPrimaryTypes()
+        {
+            var person1 = new JsonObject
+            {
+                { "name", "John" },
+                { "age", 45 },
+                { "is_married", true }
+            };
+
+            // Assign by creating a new instance of primary Json type
+            person1["name"] = new JsonString("Bob");
+
+            Assert.IsType<JsonString>(person1["name"]);
+            Assert.Equal("Bob", person1["name"] as JsonString);
+
+            // Assign by using an implicit operator on primary Json type
+            JsonNumber newAge = 55;
+            person1["age"] = newAge;
```

Comment:
Unless the JsonNumber object is different when created implicitly vs new vs explicit cast (which it shouldn't), I don't think the comment or style variation really provides value.

Reply:
Ok, it was for showing different scenarios, but as now different styles of creating JsonString, JsonBoolean and JsonNumber are tested in their unit tests, I can unify those.